### ISSUE TITLE
deps_cpp_info and cpp_info not consistent between conan install and conan build

### ISSUE THREAD

**Issue by abuszta** [conan-io/conan#590]
I have two packages:

```
from conans import ConanFile

class package1(ConanFile):
    name = 'package1'
    version = '0.1.0'

    def build(self):
        self.output.info("package1 rootpath: " + str(self.cpp_info.rootpath))
        self.output.info("package1 includepath: " + str(self.cpp_info.includedirs))
```

and

```
from conans import ConanFile

class package2(ConanFile):
    name = 'package2'
    version = '0.1.0'
    requires = 'package1/0.1.0@abusz/testing'

    def build(self):
        self.output.info("package1 rootpath: " + str(self.deps_cpp_info['package1'].rootpath))
        self.output.info("package1 includedirs: " + str(self.deps_cpp_info['package1'].includedirs))

        self.output.info("package2 rootpath: " + str(self.cpp_info.rootpath))
        self.output.info("package2 includedirs: " + str(self.cpp_info.includedirs))
```

When I export these two recipes and do "conan install package2/0.1.0@abusz/testing --build" everything works:

```
package1/0.1.0@abusz/testing: package1 rootpath: /Users/abusz/.conan/data/package1/0.1.0/abusz/testing/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9
package1/0.1.0@abusz/testing: package1 includepath: ['include']
...
package2/0.1.0@abusz/testing: package1 rootpath: /Users/abusz/.conan/data/package1/0.1.0/abusz/testing/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9
package2/0.1.0@abusz/testing: package1 includedirs: ['include']
package2/0.1.0@abusz/testing: package2 rootpath: /Users/abusz/.conan/data/package2/0.1.0/abusz/testing/package/ece115d6e9f37d18b18b2ba1b5fe503efae713e4
package2/0.1.0@abusz/testing: package2 includedirs: ['include']
```

However when I try to use "conan install /path/to/package2 --build" to get dependencies and then "conan build /path/to/package2" to build it I get this:

```
Project: package1 rootpath: 
Project: package1 includedirs: [u'/Users/abusz/.conan/data/package1/0.1.0/abusz/testing/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include']
ERROR: Unable to build it successfully
  File "/Users/abusz/tmp3/package2/conanfile.py", line 12, in build
    self.output.info("package2 rootpath: " + str(self.cpp_info.rootpath))
AttributeError: 'NoneType' object has no attribute 'rootpath'

```

I see two issues here:
- rootpath for dependencies is not populated when using conan build
- cpp_info variables (rootpath, includedirs etc.) are not available at all when using conan build

Is it possible to access that stuff in conan build? As a package maintainer I would find this "build" option very useful as I don't need to export package all the time when developing recipe.

Tested on conan 0.14.1.

**Comment by memsharded** [comment]
Quick tip: ``deps_cpp_info`` will be populated in build if you use the ``txt`` generator and ``deps_env_info`` will be populated in build if you use the ``env`` generator. This is because the "build" is not allowed to change the state of the ``conan install`` so all the information of the install has to be stored somewhere.

It is true that ``cpp_info`` is not accessible, but it shouldn't be necessary to build your own_package at all. By definition ``cpp_info`` is the information that you provide to the consumers of the package, but all the information you need to build the package should be encoded in the settings, options, and the information collected from dependencies, i.e. ``deps_xxx_info``.

If you don't want to add those generators in your conanfile recipe, just use them in the command line:

```bash
$ conan install .. -g=env -g=txt
```

**Comment by abuszta** [comment]
I'm totally ok with using env and txt generator. However despite using them I still don't have access to e.g. self.deps_cpp_info['package1'].rootpath. So I need to hack it using something like self.deps_cpp_info['package1'].includedirs + ../

**Comment by memsharded** [comment]
Ok, yes, you are right. I will check if it is possible to add it, though maybe it is not simple as it is not being serialized to disk.

**Comment by memsharded** [comment]
In any case, one question out of curiosity, why do you need to access the "rootpath" of one of your dependencies? Maybe there is an interesting use case that could have a better solution or something we could improve.

**Comment by memsharded** [comment]
Oh, yes, that makes sense. In fact we had a ``res`` folder to store resources, data, etc., but it is not very used, I think that it is not serialized to ``conanbuildinfo.txt``. Actually I remember now that adding a ``rootpath`` was partly to let users use their own folders without having to hardcode them in conan.

I have submitted a PR, it will fix it, but it has to pass CI, which has been broken these days, I think due to the large DDoS attacks. Re-running it, it will be released in next 0.15, please keep using the workaround until then :)

**Comment by memsharded** [comment]
Fixed, will be released in next conan 0.15.

**Comment by abuszta** [comment]
I have package with some additional files needed for building (e.g for cmake) which do not fit into one of available categories (include, bin, lib dirs). I just decided to export this files into root of dependent package :)
